Sources/EitherSingleOrList.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  EitherSingleOrList.swift
+//  InstantSearchCore-iOS
+//
+//  Created by Vladislav Fitc on 04/03/2019.
+//  Copyright © 2019 Algolia. All rights reserved.
+//
+
+import Foundation
+
+enum EitherSingleOrList<T> {
+    
+    case single(T)
+    case list([T])
+    
+    init(_ v: T) {
+        self = .single(v)
+    }
+    
+    init<S: Sequence>(_ v: S) where S.Element == T {
+        self = .list(Array(v))
+    }
+    
+}
+
+extension EitherSingleOrList: CustomStringConvertible {
+    
+    var description: String {
+        switch self {
+        case .single(let value):
+            return (value as? CustomStringConvertible)?.description ?? "\(value)"
+            
+        case .list(let list):
+            return list.description
+        }
+    }
+    
+}
+
+extension EitherSingleOrList: Codable where T: Codable {
```

Comment:
Maybe I misunderstood, but did we want to change that to `SingleOrList`?

Reply:
Yes, in the end we decided to keep EitherSingleOrList. But I'm not a big fan of it, can rename if you assume that it's better.

Reply:
As this is not public, I'm really fine with keeping it as well. It's understandable, so I'm good with the name